plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -0,0 +1,207 @@
+
+    standalone_logger.info("Bootstrapping %s: %s" % (manifest.engine_name, message))
+
+
+def handle_bootstrap_completed(engine):
+    """
+    Callback function that handles cleanup after successful completion of the bootstrap.
+
+    :param engine: Launched :class:`sgtk.platform.Engine` instance.
+    """
+
+    # Needed global to re-import the toolkit core.
+    global sgtk
+
+    # Re-import the toolkit core to ensure usage of a swapped in version.
+    import sgtk
+
+    # Add a logout menu item to the engine context menu.
+    sgtk.platform.current_engine().register_command(ITEM_LABEL_LOGOUT,
+                                                    logout_user,
+                                                    {"type": "context_menu"})
+
+
+def handle_bootstrap_failed(step, exception):
+    """
+    Callback function that handles cleanup after failed completion of the bootstrap.
+
+    :param step: Bootstrap step ("sgtk" or "engine") that raised the exception.
+    :param exception: Python exception raised while bootstrapping.
+    """
+
+    # Needed global to re-import the toolkit core.
+    global sgtk
+
+    if step == "engine":
+        # Re-import the toolkit core to ensure usage of a swapped in version.
+        import sgtk
+
+    # Report the encountered exception.
+    standalone_logger.info("Bootstrapping %s failed: %s" % (manifest.engine_name, exception))
+
+    # Clear the user's credentials to log him/her out.
+    sgtk.authentication.ShotgunAuthenticator().clear_default_user()
+
+    # Re-display the login menu.
+    create_login_menu()
+
+
+def logout_user():
+    """
+    Shuts down the engine and logs out the user of Shotgun.
+    """
+
+    # Shutting down the engine also get rid of the engine menu.
+    plugin_engine.shutdown()
+
+    # Clear the user's credentials to log him/her out.
+    sgtk.authentication.ShotgunAuthenticator().clear_default_user()
+
+    # Re-display the login menu.
+    create_login_menu()
+
+
+def create_login_menu():
+    """
+    Creates and displays a Shotgun user login menu.
+    """
+
+    # Creates the menu entry in the application menu bar.
+    menu = pm.menu(MENU_LOGIN, label=MENU_LABEL, parent=pm.melGlobals["gMainWindow"])
+
+    # Add the login menu item.
+    pm.menuItem(parent=menu, label=ITEM_LABEL_LOGIN, command=pm.Callback(login_user))
+
+    # Add the website menu item.
+    pm.menuItem(parent=menu, label=ITEM_LABEL_WEBSITE, command=pm.Callback(jump_to_website))
+
+
+def delete_login_menu():
+    """
+    Deletes the displayed Shotgun user login menu.
+    """
+
+    if pm.menu(MENU_LOGIN, exists=True):
+        pm.deleteUI(MENU_LOGIN)
+
+
+def jump_to_website():
+    """
+    Jumps to the Shotgun website in the defaul web browser.
+    """
+
+    # Import Qt locally from PySide since sgtk.platform.qt only works after engine initialization.
+    from PySide import QtCore, QtGui
+
+    QtGui.QDesktopServices.openUrl(QtCore.QUrl(WEBSITE_URL))
```

Comment:
Should this link be to: https://www.shotgunsoftware.com/signup/ assuming it is to get new folks to sign up?